source/fundamentals/gridfs.txt

```diff
@@ -0,0 +1,395 @@
+======
+GridFS
+======
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+This guide shows you how to store and retrieve large files in MongoDB using
+**GridFS**. GridFS is a specification implemented by the driver that describes
+how to split files into chunks when storing them and reassemble them when
+retrieving them. The driver implementation of GridFS is an abstraction that
+manages the operations and organization of the file storage.
+
+You should use GridFS if the size of your files exceed the BSON document
+size limit of 16MB. For more detailed information on whether GridFS is
+suitable for your use case, see the :manual:`GridFS server manual page </core/gridfs>`.
+
+See the following sections that describe GridFS operations and how to
+perform them:
+
+- :ref:`Create a GridFS bucket <gridfs-create-bucket>`
+- :ref:`Store Files <gridfs-store-files>`
+- :ref:`Retrieve File Information <gridfs-retrieve-file-info>`
+- :ref:`Download Files <gridfs-download-files>`
+- :ref:`Rename Files <gridfs-rename-files>`
+- :ref:`Delete Files <gridfs-delete-files>`
+- :ref:`Delete a GridFS bucket <gridfs-delete-bucket>`
+
+How GridFS Works
+----------------
+
+GridFS organizes files in a **bucket**, a group of MongoDB collections
+that contain the chunks of files and information describing them. The
+bucket contains the following collections, named using the convention
+defined in the GridFS specification:
+
+- The ``chunks`` collection stores the binary file chunks.
+- The ``files`` collection stores the file metadata.
+
+When you create a new GridFS bucket, the driver creates the collections
+referenced above, prefixed with the default bucket name ``fs``, unless
+you specify a different name. The driver also creates an index on each
+collection to ensure efficient retrieval of the files and related
+metadata. For more information on GridFS indexes, see the server manual
```

Comment:
Note: Indexes will only be added if the files collection doesn't exist or is empty.

See: https://github.com/mongodb/specifications/blob/master/source/gridfs/gridfs-spec.rst#before-write-operations